Userland/Libraries/LibGUI/UndoStack.cpp

```diff
@@ -59,7 +59,7 @@ void UndoStack::push(NonnullOwnPtr<Command>&& command)
         finalize_current_combo();
 
     if (m_stack_index > 0) {
-        for (size_t i = 0; i < m_stack_index; i++)
+        for (size_t i = 0; i < m_stack_index; i++, m_clean_index--)
```

Comment:
This doesn't seem right. Why does the clean index change when pushing something on the undo stack?
The clean index should stay the same until we save the document (and mark the undo stack clean at its current location.)

Reply:
The clean index changes because the m_stack_index is counted backwards.
Every time we push a command it is added to the beginning of the vector and I have to correct the clean index to stay where it was.
For example, the scenario in your comment below is not correct because as you add commands to the stack, the stack_index is always zero. When we undo the stack_index increments.